flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/calcite/RelTimeIndicatorConverter.scala

```diff
@@ -602,7 +602,7 @@ class RexTimeIndicatorMaterializerUtils(rexBuilder: RexBuilder) {
   private def timestamp(isNullable: Boolean): RelDataType = rexBuilder
     .getTypeFactory
     .asInstanceOf[FlinkTypeFactory]
-    .createTypeFromInternalType(InternalTypes.TIMESTAMP, isNullable = isNullable)
+    .createTypeFromLogicalType(new TimestampType(), isNullable = isNullable)
```

Comment:
Shouldn't the nullability only set in the logical type now and not be a separate flag in this method?

Reply:
I think this PR is big enough. My idea is to override the functionality of `InternalType` first, and then to extend the functionality(like nullable) of `LogicalType` in the next PR.